### PR TITLE
support for Jakarta persistence layer (JPL) annotations 

### PR DESCRIPTION
**Pull request by jgaleotti**
I am having problems having Spring2 and Spring3 modules simultaneously.

### CODE REVIEW

File: e2e-tests/spring-rest-jakarta/pom.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>io.springfox</groupId>
+            <artifactId>springfox-swagger2</artifactId>
```
Comment (arcuri82) [inline e2e-tests/spring-rest-jakarta/pom.xml:59]: better to configure SpringDoc, as SpringFox is kind of dead...

File: client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/ClassAnalyzer.java [hunk 2]
```diff
@@ -11,22 +11,84 @@
...
+    private static final String JAKARTA_DIGITS_ANNOTATION_NAME = "jakarta.validation.constraints.Digits";
+    private static final String JAVAX_ENTITY_ANNOTATION_NAME = "javax.persistence.Entity";
+    private static final String JAKARTA_ENTITY_ANNOTATION_NAME = "jakarta.persistence.Entity";
+    private static final String JAVAX_TABLE_ANNOTATION_NAME = "javax.persistence.Table";
+    private static final String JAKARTA_TABLE_ANNOTATION_NAME = "jakarta.persistence.Table";
+
+    private static final String JAVAX_NOT_NULL_ANNOTATION_NAME = "javax.validation.constraints.NotNull";
+    private static final String JAKARTA_NOT_NULL_ANNOTATION_NAME = "jakarta.validation.constraints.NotNull";
+    public static final String JAVAX_COLUMN_ANNOTATION_NAME = "javax.persistence.Column";
+    public static final String JAKARTA_COLUMN_ANNOTATION_NAME = "jakarta.persistence.Column";
```
Comment (arcuri82) [inline client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/ClassAnalyzer.java:63]: why are those `public` while the others are `private`?

File: client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/ClassAnalyzer.java [hunk 3]
```diff
@@ -11,22 +11,84 @@
...
+        boolean canUseJavaPersistenceApi = canUseJavaxJPA();
+
+        // Jakarta Persistence
+        boolean useJakartaPersistenceLayer;
+        if (canUseJavaPersistenceApi) {
+            // If JPA can be used, then JPL cannot be used
+            useJakartaPersistenceLayer = false;
+        } else {
+            // Check if Jakarta Persistence Layer can be used
+            useJakartaPersistenceLayer = canUseJakartaPersistenceLayer();
```
Comment (arcuri82) [inline client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/ClassAnalyzer.java:90]: naming should be consistent. if we have `canUseJavaPersistenceApi`, then I guess should be `canUseJakartaPersistenceLayer`?

File: client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/ClassAnalyzer.java [hunk 4]
```diff
@@ -254,20 +368,39 @@ private static Long getMaxValue(Field f) throws Exception {
...
+        final String patternAnnotationName = getAnnotationName(namespace, JAVAX_PATTERN_ANNOTATION_NAME, JAKARTA_PATTERN_ANNOTATION_NAME);
+        return getStringElement(f, patternAnnotationName, "regexp");
     }
 
-    private static String getPatterRegExp(Field f) throws Exception {
-        return getStringElement(f, "javax.validation.constraints.Pattern", "regexp");
+    private static String getAnnotationName(NameSpace namespace, String javaxAnnotationName, String jakartaAnnotationName) {
+        switch (namespace) {
+            case JAVAX:
+                return javaxAnnotationName;
```
Comment (arcuri82) [inline client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/ClassAnalyzer.java:398]: need "assert" here (as it is a private method, and not a public one). assert that `javaxAnnotationName` starts with `javax.`, and other way round for `jakartaAnnotationName`

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/ClassAnalyzer.java (hunk 2 above)
Comment (jgaleotti) [reply client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/ClassAnalyzer.java:63]: no reason, simply a mistake.

File: client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/ClassAnalyzer.java (hunk 3 above)
Comment (jgaleotti) [reply client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/ClassAnalyzer.java:90]: my bad.

File: client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/ClassAnalyzer.java (hunk 4 above)
Comment (jgaleotti) [reply client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/ClassAnalyzer.java:398]: ack

File: e2e-tests/pom.xml [hunk 5]
```diff
@@ -68,4 +68,16 @@
         </pluginManagement>
     </build>
 
+    <profiles>
```
[the remark this answers is not in the document]
Comment (jgaleotti) [reply e2e-tests/pom.xml:71]: ok

File: e2e-tests/spring-rest-jakarta/pom.xml (hunk 1 above)
Comment (jgaleotti) [reply e2e-tests/spring-rest-jakarta/pom.xml:59]: ok

**Review by arcuri82 — APPROVED** [review]

**Pull request merged**